### PR TITLE
fix tagged pointer c++ tests for asan/non asan cases.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Review by jvoisin — APPROVED** [review]

**Comment by struct** [comment]
Thanks! Can you rebase? I just merged a new CI check for `make-format`

File: tests/tagged_ptr_test.cpp [hunk 1]
```diff
@@ -100,18 +101,21 @@ int main(int argc, char *argv[]) {
             abort();
         }
 
+#if defined(__SANITIZE_ADDRESS__)
```
Comment (struct) [inline tests/tagged_ptr_test.cpp:104]: Does `ENABLE_ASAN` not work here? Its already set by the Makefile

Comment (devnexen) [reply tests/tagged_ptr_test.cpp:104]: it possibly can, but at least here we more certain if it comes from the compiler ?

Comment (struct) [reply tests/tagged_ptr_test.cpp:104]: True but that probably means there should be a different check to verify that `address_sanitizer` and `memory_sanitizer` are present when ENABLE_ASAN is and that probably does not need to be here in a test.

File: include/compiler.h [hunk 2]
```diff
@@ -28,3 +28,12 @@
 
 #define LIKELY(x) __builtin_expect(!!(x), 1)
 #define UNLIKELY(x) __builtin_expect(!!(x), 0)
+
+#if __clang__
```
Comment (struct) [inline include/compiler.h:32]: I don't think we need these now. But this is actually a good place to check if these are present and `ENABLE_ASAN` is false. We should probably put a static_assert in there for that. WDYT?

**Pull request merged**